contract_api/registry.py

```diff
@@ -0,0 +1,273 @@
+                        "service_id": all_orgs_srvcs[rec]['service_id'],
+                        "members": all_orgs_members.get(rec, [])}
+                all_orgs_data.append(data)
+            return all_orgs_data
+        except Exception as e:
+            print(repr(e))
+            raise e
+
+    def _prepare_subquery(self, s, q, fm):
+        try:
+            sub_qry = ""
+            if s == "all":
+                for rec in fm:
+                    sub_qry += fm[rec] + " LIKE '" + str(q) + "%' OR "
+                sub_qry = sub_qry[:-3] if sub_qry.endswith("OR ") else sub_qry
+            else:
+                sub_qry += fm[s] + " LIKE '" + str(q) + "%' "
+            return sub_qry.replace("org_id", "M.org_id")
+        except Exception as err:
+            raise err
+
+    def _get_total_count(self, sub_qry, filter_query, values):
+        try:
+            if filter_query != "":
+                filter_query = " AND " + filter_query
+            search_count_query = "SELECT count(*) as search_count FROM service A, (SELECT DISTINCT M.org_id, M.service_id FROM " \
+                                 "service_metadata M LEFT JOIN service_tags T ON M.service_row_id = T.service_row_id WHERE (" \
+                                 + sub_qry.replace('%', '%%') + ")" + filter_query + ") B WHERE A.service_id = B.service_id " \
+                                 "AND A.org_id = B.org_id AND A.is_curated = 1 "
+
+            res = self.repo.execute(search_count_query, values)
+            return res[0].get("search_count", 0)
+        except Exception as err:
+            raise err
+
+    def _search_query_data(self, sub_qry, sort_by, order_by, offset, limit, filter_query, values):
+        try:
+            if filter_query != "":
+                filter_query = " AND " + filter_query
+            srch_qry = "SELECT * FROM service A, (SELECT M.org_id, M.service_id, group_concat(T.tag_name) AS tags FROM " \
+                       "service_metadata M LEFT JOIN service_tags T ON M.service_row_id = T.service_row_id WHERE " \
+                       + sub_qry.replace('%', '%%') + filter_query + " GROUP BY M.org_id, M.service_id ORDER BY %s %s ) B WHERE " \
+                "A.service_id = B.service_id AND A.org_id=B.org_id AND A.is_curated= 1 LIMIT %s , %s"
+
+            qry_dta = self.repo.execute(
+                srch_qry, values + [sort_by, order_by, int(offset), int(limit)])
+            org_srvc_tuple = ()
+            rslt = {}
+            for rec in qry_dta:
+                org_id = rec['org_id']
+                service_id = rec['service_id']
+                tags = rec['tags']
+                org_srvc_tuple = org_srvc_tuple + ((org_id, service_id),)
+                if org_id not in rslt.keys():
+                    rslt[org_id] = {}
+                if service_id not in rslt[org_id].keys():
+                    rslt[org_id][service_id] = {}
+                rslt[org_id][service_id]["tags"] = tags
```

Comment:
The in clause should be bounded. If you we end up with 10,000 services qualifying this IN query can explode. We should be looking to split into multiple queries beyond a certain size of the qry_dta

Reply:
Created Issue #156. Will fix this in near time